checkov/terraform/graph_builder/foreach/abstract_handler.py

```diff
@@ -1,173 +1,91 @@
-        foreach_statement = evaluate_terraform(block.attributes.get(FOREACH_STRING))
-        count_statement = evaluate_terraform(block.attributes.get(COUNT_STRING))
-        if foreach_statement:
-            return self._is_static_foreach_statement(foreach_statement)
-        if count_statement:
-            return self._is_static_count_statement(count_statement)
-        return False
-
-    @staticmethod
-    def extract_from_list(val: list[str] | list[int]) -> list[str] | list[int] | int | str:
-        return val[0] if len(val) == 1 and isinstance(val[0], (str, int)) else val
-
-    def _handle_static_foreach_statement(self, statement: list[str] | dict[str, Any]) -> Optional[list[str] | dict[str, Any]]:
-        if isinstance(statement, list):
-            statement = self.extract_from_list(statement)
-        evaluated_statement = evaluate_terraform(statement)
-        if isinstance(evaluated_statement, str):
-            try:
-                evaluated_statement = json.loads(evaluated_statement)
-            except ValueError:
-                pass
-        if isinstance(evaluated_statement, set):
-            evaluated_statement = list(evaluated_statement)
-        if isinstance(evaluated_statement, (dict, list)) and all(isinstance(val, str) for val in evaluated_statement):
-            return evaluated_statement
-        return
-
-    def _handle_static_count_statement(self, statement: list[str] | int) -> Optional[int]:
+    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], block_idx: int) -> None:
+        main_resource = self.local_graph.vertices[block_idx]
         if isinstance(statement, list):
-            statement = self.extract_from_list(statement)
-        evaluated_statement = evaluate_terraform(statement)
-        if isinstance(evaluated_statement, int):
-            return evaluated_statement
-        return
+            for i, new_value in enumerate(statement):
+                self._create_new_foreach_resource(block_idx, i, main_resource, new_key=new_value, new_value=new_value)
+        if isinstance(statement, dict):
+            for i, (new_key, new_value) in enumerate(statement.items()):
+                self._create_new_foreach_resource(block_idx, i, main_resource, new_key, new_value)
 
-    def _handle_static_statement(self, block_index: int, sub_graph: Optional[l_graph.TerraformLocalGraph] = None) -> Optional[list[str] | dict[str, Any] | int]:
-        attrs = self.local_graph.vertices[block_index].attributes if not sub_graph else sub_graph.vertices[block_index].attributes
-        foreach_statement = attrs.get(FOREACH_STRING)
-        count_statement = attrs.get(COUNT_STRING)
-        if foreach_statement:
-            return self._handle_static_foreach_statement(foreach_statement)
-        if count_statement:
-            return self._handle_static_count_statement(count_statement)
-        return
+    @abc.abstractmethod
```

Comment:
We should probably place all abstractmethods together at the beginning of the class